pkg/channel/message_dispatcher.go

```diff
@@ -189,11 +189,15 @@ func (d *MessageDispatcherImpl) executeRequest(ctx context.Context, url *url.URL
 		return ctx, nil, nil, &execInfo, err
 	}
 
-	kncloudevents.SetAcceptReplyHeader(additionalHeaders, kncloudevents.AcceptReplyResponse)
+	headers := nethttp.Header{}
+	for k, v := range additionalHeaders {
+		headers[k] = v
+	}
+	kncloudevents.SetAcceptReplyHeader(headers, kncloudevents.AcceptReplyResponse)
```

Comment:
Since this header is pretty much static, would it be better to set it in the channel receiver, so you don't need to realloc this map every time? I'm talking about line 181 of message_receiver.go

Reply:
I don't like new map either. But putting it in receiver seems problematic. For example, the handler can choose to react to this header from upstream sender too and adding this to all the incoming message might confuse the message handler. So I am trying to add/overwrite it as late as possible for downstream only. WDYT?

Reply:
But you're doing the same here right? You're adding this header to all the dispatched messages anyway?

Reply:
Yes, it should be in all dispatched messages from the brokers. But maybe I didn't understand your original suggestion. Are you suggesting to change the following to sth like: 
```
headers := utils.PassThroughHeaders(request.Header)
kncloudevents.SetAcceptReplyHeader(headers, kncloudevents.AcceptReplyResponse)
err = r.receiverFunc(....)
```

https://github.com/knative/eventing/blob/010aec0627868981a9d6c7c81517e75517ee46f5/pkg/channel/message_receiver.go#L181

Reply:
yes, while you remove the code here

Reply:
I have looked into how the receiver and dispatch functions are used. IIUC, they are mostly bundled together in the Fanout, the only exception is that the inmemorychannel controller actually starts a dispatcher though the fanout is used in the reconciler for per channel. Is this a concern?

https://github.com/knative/eventing/blob/1c9d6c3550010f796b5b48d06c9cb8b920453676/pkg/reconciler/inmemorychannel/dispatcher/controller.go#L135

Reply:
Fixed as suggested.